cpp/src/structs/structs_column_factories.cu

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <algorithm>
+
+namespace cudf
+{
+  namespace 
+  {
+    // Helper function to superimpose validity of parent struct
+    // over all member fields (i.e. child columns).
+    void superimpose_validity(
+      rmm::device_buffer const& parent_null_mask, 
+      size_type parent_null_count,
+      std::vector<std::unique_ptr<column>>& children,
+      cudaStream_t stream,
+      rmm::mr::device_memory_resource* mr
+    )
+    {
+      if (parent_null_mask.is_empty()) {
+        // Struct is not nullable. Children do not need adjustment.
+        // Bail.
+        return;
+      }
+
+      std::for_each(
+        children.begin(),
+        children.end(),
+        [&](std::unique_ptr<column>& p_child)
```

Comment:
Capturing everything makes it difficult to see what dependencies are being injected.

Reply:
Agreed. I'll tighten this up.